In [1]:
import pandas as pd

# 1) Load the dataset

In [2]:
df = pd.read_csv("consolidated_coin_data.csv")

# 2) Standardize column names

In [3]:
df.columns = (
    df.columns.str.strip()
              .str.lower()
              .str.replace(" ", "_")
)

# Expected final column names:
# ['currency', 'date', 'open', 'high', 'low', 'close', 'volume', 'market_cap']

# 3) Remove NULL values

In [4]:
df = df.dropna()

# 4) Convert numeric columns
# Remove commas → convert to float

In [5]:
numeric_cols = ["open", "high", "low", "close", "volume", "market_cap"]

for col in numeric_cols:
    df[col] = (
        df[col].astype(str)
               .str.replace(",", "", regex=False)
               .astype(float)
    )

# 5) Convert Date column

In [6]:
df["date"] = pd.to_datetime(df["date"], format="%d-%b-%y")

# 6) Sort values correctly

In [7]:
df = df.sort_values(by=["currency", "date"]).reset_index(drop=True)

# 7) Create additional columns for Power BI

In [8]:
df["year"] = df["date"].dt.year
df["month"] = df["date"].dt.month
df["year_month"] = df["date"].dt.to_period("M").astype(str)

# 8) Save cleaned dataset

In [9]:
df.to_csv("cleaned_coin_data.csv", index=False)

In [10]:
df.head()

currency       date  open  high   low  close  volume  market_cap  year  \
0  binance-coin 2013-04-28  4.30  4.40  4.18   4.35     0.0  74636938.0  2013   
1  binance-coin 2013-04-29  4.37  4.57  4.23   4.38     0.0  75388964.0  2013   
2  binance-coin 2013-04-30  4.40  4.57  4.17   4.30     0.0  74020918.0  2013   
3  binance-coin 2013-05-01  4.29  4.36  3.52   3.80     0.0  65604596.0  2013   
4  binance-coin 2013-05-02  3.78  4.04  3.01   3.37     0.0  58287979.0  2013   

   month year_month  
0      4    2013-04  
1      4    2013-04  
2      4    2013-04  
3      5    2013-05  
4      5    2013-05

In [11]:
df.count()

currency      28944
date          28944
open          28944
high          28944
low           28944
close         28944
volume        28944
market_cap    28944
year          28944
month         28944
year_month    28944
dtype: int64

### Count of version of the CSV was **28944** and after cleaning it up we have **28944**. So we had no NULL rows